test/phoenix/router/forward_test.exs

```diff
@@ -0,0 +1,130 @@
+defmodule Phoenix.Router.ForwardTest do
+  use ExUnit.Case, async: true
+  use RouterHelper
+
+  defmodule Controller do
+    use Phoenix.Controller
+
+    plug :assign_fwd_conn
+
+    def index(conn, _params), do: text(conn, "admin index")
+    def stats(conn, _params), do: text(conn, "stats")
+    def api_users(conn, _params), do: text(conn, "api users")
+    def api_root(conn, _params), do: text(conn, "api root")
+    defp assign_fwd_conn(conn, _), do: assign(conn, :fwd_conn, conn)
```

Comment:
Do you need to do this? Can't you just use the conn return itself? I think the private information in the `conn` return and the one assigned is the same, no?